### DISCUSSION TITLE
Can freeCAD convert .dwg to any image format via python code?

### DISCUSSION THREAD

**Discussion by ufoozhenghao** [FreeCAD/FreeCAD#11723] [Code contribution discussions]
I want a function that when i send a .dwg file to pyhon, it can convert it to image. Seems that it can be done in gui? 

And the wiki says i can view python code, but i only found c code in github respo. Can anyone show me the pyhton code link?

**Comment by yorikvanhavre** [comment]
You can load a dwg and generate an image without using the GUI. Have a look at https://github.com/FreeCAD/FreeCAD/blob/main/src/Mod/Draft/importDWG.py and https://github.com/FreeCAD/FreeCAD/blob/main/src/Mod/Arch/OfflineRenderingUtils.py

About C++ and Python, most of FreeCAD is coded in C++, but several modules in src/Mod are coded in Python, and also all the addon modules at https://github.com/FreeCAD/FreeCAD-addons

**Comment by ufoozhenghao** [comment]
thx

Xuan
________________________________
